alibi_detect/tests/test_dep_management.py

```diff
@@ -0,0 +1,274 @@
+    """
+
+    dependency_map = defaultdict(lambda: ['default'])
+    for dependency, relations in [
+        ('fetch_detector', ['tensorflow']),
+        ('fetch_tf_model', ['tensorflow'])
+    ]:
+        dependency_map[dependency] = relations
+    from alibi_detect.utils import fetching
+    check_correct_dependencies(fetching, dependency_map, opt_dep)
+
+
+def test_saving_tf_dependencies(opt_dep):
+    """Tests that the alibi_detect.saving.tensorflow module correctly protects against uninstalled optional
+    dependencies.
+    """
+
+    dependency_map = defaultdict(lambda: ['default'])
+    for dependency, relations in [
+        ('Detector', ['tensorflow']),
+        ('load_detector_legacy', ['tensorflow']),
+        ('load_embedding_tf', ['tensorflow']),
+        ('load_kernel_config_tf', ['tensorflow']),
+        ('load_model_tf', ['tensorflow']),
+        ('load_optimizer_tf', ['tensorflow']),
+        ('prep_model_and_emb_tf', ['tensorflow']),
+        ('save_detector_legacy', ['tensorflow']),
+        ('save_model_config_tf', ['tensorflow']),
+        ('get_tf_dtype', ['tensorflow'])
+    ]:
+        dependency_map[dependency] = relations
+    from alibi_detect.saving import tensorflow as tf_saving
+    check_correct_dependencies(tf_saving, dependency_map, opt_dep)
+
+
+def test_saving_dependencies(opt_dep):
+    """Tests that the alibi_detect.saving module correctly protects against uninstalled optional dependencies."""
+
+    dependency_map = defaultdict(lambda: ['default'])
+    for dependency, relations in []:
+        dependency_map[dependency] = relations
+    from alibi_detect import saving
+    check_correct_dependencies(saving, dependency_map, opt_dep)
+
+
+def test_tensorflow_utils_dependencies(opt_dep):
+    """Tests that the saving utils module correctly protects against uninstalled optional dependencies.
+    """
+
+    dependency_map = defaultdict(lambda: ['default'])
+    for dependency, relations in [
+            ("batch_compute_kernel_matrix", ['tensorflow']),
+            ("mmd2", ['tensorflow']),
+            ("mmd2_from_kernel_matrix", ['tensorflow']),
+            ("relative_euclidean_distance", ['tensorflow']),
+            ("squared_pairwise_distance", ['tensorflow']),
+            ("GaussianRBF", ['tensorflow']),
+            ("DeepKernel", ['tensorflow']),
+            ("permed_lsdds", ['tensorflow']),
+            ("predict_batch", ['tensorflow']),
+            ("predict_batch_transformer", ['tensorflow']),
+            ("quantile", ['tensorflow']),
+            ("subset_matrix", ['tensorflow']),
+            ("zero_diag", ['tensorflow']),
+            ("mutate_categorical", ['tensorflow']),
+            ("TFDataset", ['tensorflow'])
+            ]:
+        dependency_map[dependency] = relations
+    from alibi_detect.utils import tensorflow as tensorflow_utils
+    check_correct_dependencies(tensorflow_utils, dependency_map, opt_dep)
+
+
+def test_pytorch_utils_dependencies(opt_dep):
```

Comment:
The other test functions are all named `test_torch_...` rather than `test_pytorch_...`. Change for consistency?